base/linalg/symmetric.jl

```diff
@@ -107,6 +107,21 @@ end
     r
 end
 
+function setindex!(A::Symmetric, v, i::Integer, j::Integer)
+    i == j || throw(ArgumentError("Cannot set a non-diagonal index in a symmetric matrix"))
```

Comment:
?. This is a symmetric wrapper so it is implicitly symmetric no matter what you do with it. Setting `A[i,j] = v` is the same as `A[i,j] = A[j,i] = v`.